File: leetcode/by_level/easy/js/804.js
const morseList = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]

const solution = (list) => {
  let result = []
  for (let i = 0; i < list.length; i++) {
    const charList = list[i].split('')
    let wordToMorse = []
    for (let j = 0; j < charList.length; j++) {
      const charCodeNum = charList[j].charCodeAt() - 97
      const morseCode = morseList[charCodeNum]
      wordToMorse.push(morseCode)
    }
    wordToMorse = wordToMorse.join('')
    if (!result.includes(wordToMorse)) result.push(wordToMorse)
  }
  return result
}

const input = ["gin", "zen", "gig", "msg"]

console.log(solution(input))
// solution(input)

File: leetcode/easy/py/804.py
def Solution(words):
  morseList = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
  result = []
  for word in words:
    charList = list(word)
    wordToMorse = []
    for character in charList:
      charCodeNum = ord(character) - 97
      morseCode = morseList[charCodeNum]
      wordToMorse.append(morseCode)
    s = ''
    wordToMorse = s.join(wordToMorse)
    if wordToMorse not in result:
        result.append(wordToMorse)
  return len(result)

print(Solution(["gin", "zen", "gig", "msg"]))